### PR TITLE
Fix first render back button on live redirect

### PR DESCRIPTION
**Pull request by feliperenan**
Fixes https://github.com/phoenixframework/phoenix_live_view/issues/480

### Issue
The issue is that in the first render when users click on the browser back button, even though the URL changes to the previous one, the page doesn't show the previous page content. It works fine if the user clicks on browser next button and browser back button again, but it's not a good User experience.

<details><summary>GIF before fix</summary>

</details>

### Fix
To fix that, I'm making sure the new location will be registered calling the function `registerNewLocation` right after pushing state on live redirect.

### CODE REVIEW

File: assets/js/phoenix_live_view.js [hunk 1]
```diff
@@ -1134,6 +1134,7 @@ export class View {
   onLiveRedirect({to, kind}){
     this.href = to
     Browser.pushState(kind, {}, to)
+    this.liveSocket.registerNewLocation(window.location)
```
Comment (feliperenan) [inline assets/js/phoenix_live_view.js]: @snewcomer I'm not sure if that is the proper fix though. Could you review that? maybe there is a better way to fix that 🙈

Comment (snewcomer) [reply assets/js/phoenix_live_view.js]: Agreed.  Nice find and thank you for the gif repro!

Before we merge, I want to have this testable and/or in an example app.  I'll see if we can work through this by the end of the week.

**Comment by josevalim** [comment]
Thanks @feliperenan ! Btw, please do not update the priv/static files on PRs, they cause conflicts and it can also be a security issue (people can generate a malicious compiled .js file), so we cannot accept those!

**Comment by feliperenan** [comment]
ow I've never thought about it 😱 

Thanks @josevalim

**Pull request merged**